### DISCUSSION TITLE
Add voice control

### DISCUSSION THREAD

**Discussion by ivan-kulikov-dev** [OpenViva/OpenViva#20] [Ideas]
Add voice control

test offine and online voice recognition engines 

link to libs and techs:
1) Vosk (https://github.com/alphacep/vosk-unity-asr)
2) https://github.com/snakers4/silero-models

example easy to implement voice  commands

Kurumi, сome here
Kurumi, stop

example hard to implement  voice commands

Kurumi, catch a chicken
TODO ideas of hard logical voice commands

First implement  easy commands voice, and later implement  hard logical voice commands

I can test multiple libraries/technologies and provide code/report

**Comment by funmaker** [comment]
That's a cool idea. However online solutions are rather out of the question for a project like this.
If you find some reliable offline solution that we would be able to embed into the project we can definitely look into that. Just make sure it's licensing would allow us to embed it for free.

**Reply by ivan-kulikov-dev** [reply]
I was choosed https://github.com/alphacep/vosk-api now.

This is first example of using vosk in unity.

![изображение](https://user-images.githubusercontent.com/4685934/170842003-eb65d0b6-0096-4522-9c0a-9cd018d4baa1.png)

A bit of a problem with non-standard words now (Kurumi)

**Reply by ivan-kulikov-dev** [reply]
https://file.io/xI3V8f06yYlb vosk-unity-asr-for-openviva_v1



original example from https://github.com/alphacep/vosk-unity-asr.

Script Logic

The Controller GameObject has two components that are responsible for the transcription.

Voice Processor - Handles the microphone input. Microphone can be enable and disabled. Auto detection can be disabled by passing false when calling the StartRecording method.

VoskSpeechToText - Passes the microphone input into Vosk. If KeyPhrases are not assigned, the recognizer will recognize all words in the Models dictionary. The detection starts by default but can be started by hand by calling StartVoskSTT.

**Reply by ivan-kulikov-dev** [reply]
To get around this problem, should we limit the list of given character names to only well-recognized ones? And in the next releases, refine the character name recognition model and expand the list of allowed names?

**Reply by funmaker** [reply]
Limiting character names is not a good idea, that would go against the whole point of customizing viva for your character of choosing. If anything, it'd be better to came up with phrases that do not include the character name.

**Reply by ivan-kulikov-dev** [reply]
Then, apparently, the player will need to point the headset at the character and say. (Without pressing the trigger)
example:
 сome here

**Comment by ivan-kulikov-dev** [comment]
TODO try https://github.com/snakers4/silero-models    try offline text to sound, sound to text (In unity, no python, only c# or c++ libs)
